### ISSUE TITLE
Nutzung für Major-Upgrade von PostgreSQL

### ISSUE THREAD

**Issue by ledermann** [solectrus/backup-restore#5]
Von @gereons in https://github.com/orgs/solectrus/discussions/3748#discussioncomment-11487195:

> taugt der psql-Teil der Backup/Restore-Scripte um so einen Upgrade zu machen?

Ja, definitiv. In etwa so: 

1. Backup erstellen
2. `docker compose down`
3. `compose.yml` ändern (z.B. 16 -> 17)
4. Volume-Ordner von PostgreSQL ~leeren~ umbenennen und leer neu anlegen
5. `docker compose up`
6. Restore durchführen

Das ist aber nicht für jedermann geeignet. Auch behandelt das Backup-Script immer InfluxDB und PostgreSQL gemeinsam, was in dem Fall nicht ideal ist, weil man für den Vorgang unnötigerweise InfluxDB herunterfahren muss.

Es wäre also gut, wenn das Script optional auch **nur** PostgreSQL sichern/wiederherstellen könnten.

**Comment by gereons** [comment]
Gerade erfolgreich durchgeführt, mit einem geschickt platzierten `exit` in `restore.sh` damit die InfluxDB nicht angefasst wird. Downtime: vielleicht 2 Minuten.

Ich würde allerdings Punkt 4 durch "Volume-Ordner von PostgreSQL umbenennen dann leer neu anlegen" ersetzen, so hat man einen 2. Backup wenn was schiefgeht.

**Comment by ledermann** [comment]
Ja, eine Umbenennung ist auf jeden Fall besser als ein Löschen.
